client/state/index.js

```diff
@@ -149,13 +149,17 @@ export function createReduxStore( initialState = {} ) {
 
 	const middlewares = [
 		thunkMiddleware,
+		// We need the data layer middleware to be used as early
+		// as possible, before any side effects.
+		// This avoids, for example, side effect actions like analytics
+		// being dispatched more than once.
+		require( './data-layer/wpcom-api-middleware.js' ).default,
```

Comment:
Maybe we could add a test for this, that has a reasonable description but maybe just tests that it's the second item for now?
```jsx
describe( 'wpcom-api-middleware is ordered before other side-effect middleware' )
```